### PR TITLE
ID-498 filter sentry issues

### PR DESCRIPTION
**Pull request by Ghost-in-a-Jar**
This should filter out the vast majority of the issues we def dont care about.
Ticket: https://broadworkbench.atlassian.net/browse/ID-498

### CODE REVIEW

File: src/main/scala/org/broadinstitute/dsde/workbench/sam/Boot.scala [hunk 1]
```diff
@@ -44,18 +44,6 @@ import scala.jdk.CollectionConverters._
...
-  val version: Option[String] = Option(getClass.getPackage.getImplementationVersion)
-  private def initSentry(): Unit = sentryDsn.fold(logger.warn("No SENTRY_DSN found, not initializing Sentry.")) { dsn =>
-    val options = new SentryOptions()
-    options.setDsn(dsn)
-    options.setEnvironment(sys.env.getOrElse("SENTRY_ENVIRONMENT", "unknown"))
-    options.setRelease(version.getOrElse("unknown"))
-
-    Sentry.init(options)
-    logger.info("Sentry initialized")
-  }
```
Comment (Ghost-in-a-Jar) [inline src/main/scala/org/broadinstitute/dsde/workbench/sam/Boot.scala:58]: there is a call to this initSentry fn below in boot, just moving it to a companion obj

File: src/main/scala/org/broadinstitute/dsde/workbench/sam/util/Sentry.scala [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+  def initSentry(): Unit = sentryDsn.fold(logger.warn("No SENTRY_DSN found, not initializing Sentry.")) { dsn =>
+    val options = new SentryOptions()
+    options.setDsn(dsn)
+    options.setEnvironment(sys.env.getOrElse("SENTRY_ENVIRONMENT", "unknown"))
+    options.setRelease(version.getOrElse("unknown"))
+
+    options.setBeforeSend { (event: SentryEvent, hint: Hint) =>
+      event.getThrowable match {
+        case workbenchException: WorkbenchExceptionWithErrorReport if workbenchException.errorReport.statusCode.map(statusCodesToSkip.contains) => null
+        case _ => event
```
Comment (Ghost-in-a-Jar) [inline src/main/scala/org/broadinstitute/dsde/workbench/sam/util/Sentry.scala:29]: Can someone double check my logic here

Comment (Ghost-in-a-Jar) [reply src/main/scala/org/broadinstitute/dsde/workbench/sam/util/Sentry.scala:29]: if null is returned the event isnt sent to sentry btw

**Review by Shakespeared — APPROVED** [review]
lgtm

File: src/main/scala/org/broadinstitute/dsde/workbench/sam/util/Sentry.scala (hunk 2 above)
Comment (gpolumbo-broad) [reply src/main/scala/org/broadinstitute/dsde/workbench/sam/util/Sentry.scala:29]: Logic makes sense.  I'm less sure about the blanket `statusCodesToSkip`, but this looks like an easy enough thing to tune too.  And I have no idea what are the RIGHT things we should and should not be sending to Sentry yet, so just adding the mechanism here is good.

**Review by gpolumbo-broad — APPROVED** [review]

**Pull request merged**